stores/facilitiesStore.ts

```diff
@@ -326,8 +325,37 @@ export const useFacilitiesStore = defineStore('facilitiesStore', () => {
         return serverResponse
     }
 
-    function setItemsPerPage(newLimit: number) {
-        itemsPerPage.value = newLimit
+    /**
+     * Fetches a paginated list of facilities along with their total count.
+     * This function follows the same pattern as the others two fetcher.
+     */
+    async function fetchFacilitiesWithCount(
+        filters: FacilitySearchFilters
+    ): Promise<{ filteredSearchResults: Facility[], totalCount: number }> {
+        try {
+            // Object containthe list of facility and the total count.
+            const response = await graphQLClientRequestWithRetry<{
+                facilities: {
+                    facilities: Facility[]
+                    totalCount: number
+                }
+            }>(
+                gqlClient.request.bind(gqlClient),
+                GetPaginatedFacilitiesQuery,
+                { filters }
+            )
+
+            const paginatedFacilities = response.data?.facilities
+
+            // Extract nested facilities array and totalCount
+            const filteredSearchResults = paginatedFacilities?.facilities ?? []
+            const totalCount = paginatedFacilities?.totalCount ?? 0
+
+            return { filteredSearchResults, totalCount }
```

Comment:
let's make this a typescript type returned like we talked about on the backend `facilitiesResponse` or `paginatedFacilities`